src/UI/Art/SpriteArtist/TerrainSpriteSet.java

```diff
@@ -313,12 +336,37 @@ private boolean checkTileType(GameMap map, int x, int y, boolean assumeTrue)
   /**
    * Determines the base terrain type for the provided environment terrain type.
    * For example, FOREST is a tile type, but the trees sit on a plain, so for drawing
-   * purposes (esp. terrain transitions), the base tile type of FOREST is actually PLAIN.
+   * purposes (esp. terrain transitions), the base tile type of FOREST is actually GRASS.
    */
-  private static BaseTerrain getBaseTerrainType(BaseTerrain terrain)
+  private static TerrainType getBaseTerrainType(TerrainType terrain)
   {
-    int baseIndex = terrain.getBaseIndex();
+    if( null == terrainBases )
+    {
+      terrainBases = new HashMap<TerrainType, TerrainType>();
+      terrainBases.put(TerrainType.CITY, TerrainType.GRASS);
+      terrainBases.put(TerrainType.DUNES, TerrainType.GRASS);
+      terrainBases.put(TerrainType.FACTORY, TerrainType.GRASS);
+      terrainBases.put(TerrainType.AIRPORT, TerrainType.GRASS);
+      terrainBases.put(TerrainType.FOREST, TerrainType.GRASS);
+      terrainBases.put(TerrainType.HEADQUARTERS, TerrainType.GRASS);
+      terrainBases.put(TerrainType.LAB, TerrainType.GRASS);
+      terrainBases.put(TerrainType.MOUNTAIN, TerrainType.GRASS);
+      terrainBases.put(TerrainType.GRASS, TerrainType.GRASS);
+      terrainBases.put(TerrainType.ROAD, TerrainType.GRASS);
+
+      terrainBases.put(TerrainType.BRIDGE, TerrainType.SHOAL);
+      terrainBases.put(TerrainType.SEAPORT, TerrainType.SHOAL);
+      terrainBases.put(TerrainType.SHOAL, TerrainType.SHOAL);
+
+      terrainBases.put(TerrainType.REEF, TerrainType.SEA);
+      terrainBases.put(TerrainType.SEA, TerrainType.SEA);
+
+      if( TerrainType.TerrainTypeList.length != terrainBases.size())
+      {
+        System.out.println("WARNING! TerrainSpriteSet terrains do not align with Environment!");
+      }
+    }
 
-    return Environment.getTerrainTypes().get(baseIndex);
+    return terrainBases.get(terrain);
```

Comment:
This is definitely more compact than a switch/case, but it's roughly as hard to maintain.
With all the testinfo and such printed in the console now, one warning probably won't be noticed much.
Then again, it'll probably crash on returning null, so maybe that's not a problem...

Reply:
Yeah, I'd actually like to do some work cleaning up logging (or making the logging conditional, so we can see logging for specific subsystems we care about or whatever) at some point. As you say, this is also likely to be near the end of the console output before a stack trace, so other noise may not matter. I guess we could just be proactive and throw the exception ourselves here if we really wanted.

Reply:
Throwing the exception here sounds much like the approach you took with the testing; you don't get to play the game if it's currently broken; also the fail fast philosophy.

Reply:
Fair enough. I think I'll leave it as an unchecked exception though, since it'll fail anytime you try to start a game, and whether it will pass is determined at compile time; it shouldn't be possible to "release" the game with this broken.